integration/data/test_backup.py

```diff
@@ -664,6 +667,50 @@ def test_backup_volume_deletion(grpc_replica1, grpc_replica2,  # NOQA
         cleanup_replica(grpc_replica1)
         cleanup_replica(grpc_replica2)
 
+def test_backup_volume_list(grpc_replica1, grpc_replica2,  # NOQA
+                            grpc_controller, backup_targets):  # NOQA
+    offset = 0
+    length = 128
+    address = grpc_controller.address
+
+    for backup_target in backup_targets:
+        dev = get_dev(grpc_replica1, grpc_replica2,
+                      grpc_controller)
+
+        # create a regular backup
+        snap_data = random_string(length)
+        verify_data(dev, offset, snap_data)
+        snap = cmd.snapshot_create(address)
+        backup_info = create_backup(address, snap, backup_target)
+        assert backup_info["VolumeName"] == VOLUME_NAME
+        assert backup_info["Size"] == BLOCK_SIZE_STR
+        assert snap in backup_info["SnapshotName"]
+
+        # request a volume list
+        info = cmd.backup_volume_list(address, "", backup_target)
+        assert info[VOLUME_NAME]["Name"] == VOLUME_NAME
+        assert MESSAGE_TYPE_ERROR not in info[VOLUME_NAME]["Messages"]
+
+        # place badly named backup.cfg file
+        # we want the list call to return correctly and
+        # include an error message otherwise a single volume error
+        # can stop all backup volumes from showing up
+        backup_dir = os.path.join(finddir(BACKUP_DIR, VOLUME_NAME), "backups")
+        cfg = open(os.path.join(backup_dir, "backup_1234@failure.cfg"), "w")
+        cfg.close()
+        info = cmd.backup_volume_list(address, VOLUME_NAME, backup_target,
+                                      include_backup_details=True)
+        pprint(info)
+        assert "Invalid name" in info[VOLUME_NAME]["Messages"]["error"]
```

Comment:
We still need to check if the other volumes are shown? There is only one volume we have here.

Reply:
If we are displaying one, we will display the others. Since we no longer error and go through the regular control flow.

Reply:
Will look into using multiple volumes in the test.